app/Model/ShadowAttribute.php

```diff
@@ -196,6 +196,23 @@ public function beforeSave($options = array())
 
         // convert into utc and micro sec
         $this->data = $this->Attribute->ISODatetimeToUTC($this->data, $this->alias);
+
+        $trigger_id = 'shadow-attribute-before-save';
+        $isTriggerCallable = $this->isTriggerCallable($trigger_id);
+        if ($isTriggerCallable) {
+            $triggerData = $this->data;
+            $workflowErrors = [];
+            $logging = [
+                'model' => 'ShadowAttribute',
+                'action' => 'add',
+                'id' => $this->data['ShadowAttribute']['id'],
```

Comment:
This might throw an exception if we're adding a new proposal since it won't get an ID yet.